File: src/leetcode/solved-collection/3. Longest Substring Without Repeating Characters/solution.go
func lengthOfLongestSubstring(s string) int {
    n := len(s)
    result := 0
    for i := 0; i < n; i++ {
        visited := make([]bool, 256)
        for j := i; j < n; j++ {
            if visited[s[j]] {
                break
            } else {
                result = max(j - i + 1, result)
                visited[s[j]] = true
            }
        }
        visited[s[i]] = false
    }

    return result
}

func max(a, b int) int {
    if a > b {
        return a
    } else {
        return b
    }
}

File: src/leetcode/solved-collection/3. Longest Substring Without Repeating Characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)
        result = 0

        for i in range(n):
            visited = [False for i in range(256)]
            for j in range(i, n):
                if visited[ord(s[j])]:
                    break
                else:
                    result = max(j - i + 1, result)
                    visited[ord(s[j])] = True
        return result
